src/java/org/apache/cassandra/io/sstable/format/SSTableFormat.java

```diff
@@ -115,38 +123,75 @@ default boolean delete(Descriptor desc)
 
     FileTime getLastModifiedTime(Descriptor desc);
 
-    enum Type
+    default void setup(Map<String, String> options)
+    {
+    }
+
+    class Type
     {
-        //The original sstable format
-        BIG("big", BigFormat.instance);
+        private final static ImmutableList<Type> types;
+
+        static
+        {
+            Map<String, Supplier<SSTableFormat<?, ?>>> factories = DatabaseDescriptor.getSSTableFormatFactories();
+            List<Type> typesList = new ArrayList<>(factories.size());
+            factories.forEach((key, factory) -> typesList.add(new Type(typesList.size(), key.toLowerCase(), factory.get())));
+            types = ImmutableList.copyOf(typesList);
+        }
 
+        public final int ordinal;
         public final SSTableFormat<?, ?> info;
         public final String name;
+        private final static Type[] typesArray0 = new Type[0];
+
+        private static Type currentType;
 
         public static Type current()
         {
-            return CassandraRelevantProperties.SSTABLE_FORMAT_DEFAULT.getEnum(true, Type.class);
+            if (currentType != null)
+                return currentType;
+
+            String key = CassandraRelevantProperties.SSTABLE_FORMAT_DEFAULT.getString();
```

Comment:
We should use the first defined format as the default if this option is not supplied.

Reply:
https://github.com/apache/cassandra/pull/2064/commits/80ef51bf6a43b9925949fa9e74c384dae0a5d6e9